### PR TITLE
[SYCL][ESIMD] Fix an issue causing a build break when building ESIMD on Windows platform

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #include <sycl/ext/intel/esimd/detail/defines_elementary.hpp>
 #include <sycl/ext/intel/esimd/detail/memory_intrin.hpp>
 
+using uint = unsigned int;
```
Comment (aelovikov-intel) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp:19]: I don't think spoiling global namespace like this is good. Is there a standard header where it is defined?

**Review by v-klochkov — APPROVED** [review]
Looks Ok/good to me. 
Just a comment: using 'uint32_t' may be more explicit than 'unsigned'.
'uint32_t' is already used in that detail/memory_intrin.hpp now.
@kbobrovs - FYI and maybe additional comments.

File: sycl/test/esimd/regression/windows_build_test.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/experimental/esimd/memory.hpp>
+
+class Kernel;
+
+int main()
+{
+  sycl::queue q;
+  sycl::device dev = q.get_device();
+  sycl::context ctx = q.get_context();
```
Comment (v-klochkov) [inline sycl/test/esimd/regression/windows_build_test.cpp:25]: nit: 'ctx' is unused

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp (hunk 1 above)
Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp]: +1. I think this needs to be fixed

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp]: Oh, this is already fixed.

**Review by kbobrovs — APPROVED** [review]

**Comment by kbobrovs** [comment]
> Looks Ok/good to me. Just a comment: using 'uint32_t' may be more explicit than 'unsigned'. 'uint32_t' is already used in that detail/memory_intrin.hpp now. @kbobrovs - FYI and maybe additional comments.

I think using `unsigned` is OK. It is guaranteed to be at least 16 bits, and the max value returned by the `loadstoreAlignMask` is 0x7.

**Pull request merged**